src/company_cmd.cpp

```diff
@@ -870,12 +867,16 @@ CommandCost CmdCompanyCtrl(TileIndex tile, DoCommandFlag flags, uint32 p1, uint3
 			break;
 		}
 
-		case CCA_NEW_AI: { // Make a new AI company
+		case CCA_NEW_AI: { // Make new AI companies
 			if (!(flags & DC_EXEC)) return CommandCost();
 
 			if (company_id != INVALID_COMPANY && (company_id >= MAX_COMPANIES || Company::IsValidID(company_id))) return CMD_ERROR;
-			Company *c = DoStartupNewCompany(true, company_id);
-			if (c != NULL) NetworkServerNewCompany(c, NULL);
+
+			uint num_to_start = max(1U, GB(p2, 0, 4));
+			do {
+				Company *c = DoStartupNewCompany(true, company_id);
+				if (c != NULL) NetworkServerNewCompany(c, NULL);
+			} while (--num_to_start > 0);
```

Comment:
`do/while` loops in general are very difficult to read. Why not a `while` loop here? Or better, a `for` loop?